src/currency.ts

```diff
@@ -0,0 +1,14 @@
+export function SpaceCurrency(initialCurrency:number) {
+    const value = (Array as any).from(String(initialCurrency))
+        .reverse()
+        .join('')
+        .match(/\d{1,3}/g)
+        ?.reverse()
+        .map((o: string) => 
```

Comment:
Each of your variables should have a better name. I think, you can pick better name than `o`.